src/node_worker.cc

```diff
@@ -381,7 +381,8 @@ void Worker::OnThreadStopped() {
 Worker::~Worker() {
   Mutex::ScopedLock lock(mutex_);
 
-  CHECK(stopped_ || env_ == nullptr || env_->GetAsyncRequest()->IsStopped());
+  CHECK(stopped_);
+  CHECK_NULL(env_);
```

Comment:
IIRC, there was a control flow that takes to Worker destructor without nullifying `env_` , not able to figure that out now; do you know?

Reply:
@gireeshpunathil I think that would be a bug – the child thread is not allowed to exist at this point (and the next `CHECK` verifies that the thread has been joined), and the child thread in turn owns the `Environment`.